src/store/modules/position.store.js

```diff
@@ -220,7 +220,12 @@ const actions = {
                     y: centerOfExtent[1],
                 })
             }
-            const newResolution = (extent[1][0] - extent[0][0]) / rootState.ui.width
+            // taking the biggest resolution calculated with either the width or height of the screen
+            // so that it fits to the screen
+            const newResolution = Math.max(
+                (extent[1][0] - extent[0][0]) / rootState.ui.width,
+                (extent[1][1] - extent[0][1]) / rootState.ui.height
+            )
```

Comment:
Maybe as a rule of thumb taking a resolution of one lower would be good, or removing the space taken by the menu in desktop view for the ui width.